src/test/java/com/hubspot/jinjava/lib/filter/FromJsonFilterTest.java

```diff
@@ -46,21 +48,73 @@ public void itFailsWhenParameterIsNotString() {
     filter.filter(nestedJson, interpreter);
   }
 
+  @Test
+  public void itReadsEmptyJsonObjectString() {
+    JsonNode node = (JsonNode) filter.filter(EMPTY_JSON_OBJECT, interpreter);
+    assertThat(node.elements().hasNext()).isEqualTo(false);
+  }
+
+  @Test
+  public void itReadsStringAsObject() {
+    JsonNode node = (JsonNode) filter.filter(NESTED_JSON, interpreter);
+
+    checkNestedJson(node);
+  }
+
   @Test
   public void itReadsSafeStringAsObject() {
     SafeString nestedJson = new SafeString(NESTED_JSON);
-    HashMap<String, Object> node = (HashMap<String, Object>) filter.filter(
-      nestedJson,
-      interpreter
-    );
-    checkedNestJson(node);
+    JsonNode node = (JsonNode) filter.filter(nestedJson, interpreter);
+
+    checkNestedJson(node);
   }
 
-  private void checkedNestJson(HashMap<String, Object> node) {
-    assertThat(node.get("first")).isEqualTo(Arrays.asList(1, 2, 3));
+  @Test
+  public void itReadsEmptyJsonArrayString() {
```

Comment:
I think a better option for these tests is to render a jinjava template directly. See https://github.com/HubSpot/jinjava/blob/master/src/test/java/com/hubspot/jinjava/lib/filter/MinusTimeFilterTest.java#L37 as an example. That way we can make sure the API from Jinjava perspective is the same: for example: `{% set obj = test|fromjson %} {{ obj.property }}`